crates/red_knot_python_semantic/src/semantic_index/except_handlers.rs

```diff
@@ -0,0 +1,79 @@
+use std::cell::{RefCell, RefMut};
+
+use super::use_def::FlowSnapshot;
+use super::SemanticIndexBuilder;
+
+#[derive(Debug, Default)]
+pub(super) struct TryBlockContextsStack(Vec<TryBlockContexts>);
+
+impl TryBlockContextsStack {
+    pub(super) fn push_context(&mut self) {
+        self.0.push(TryBlockContexts::default());
+    }
+
+    pub(super) fn current_try_block_context(&self) -> &TryBlockContexts {
+        self.0
+            .last()
+            .expect("There should always be at least one `TryBlockContexts` on the stack")
+    }
+
+    pub(super) fn pop_context(&mut self) {
+        self.0.pop();
+    }
+}
+
+#[derive(Debug, Default)]
+pub(super) struct TryBlockContext {
+    snapshots: Vec<FlowSnapshot>,
+    visiting_nested_try_stmt: bool,
+}
+
+impl TryBlockContext {
+    pub(super) fn record_definition(&mut self, builder: &SemanticIndexBuilder) {
+        if !self.visiting_nested_try_stmt {
+            self.snapshots.push(builder.flow_snapshot());
+        }
+    }
+
+    pub(super) fn record_visiting_nested_try_stmt(&mut self) {
+        self.visiting_nested_try_stmt = true;
+    }
+
+    pub(super) fn record_exiting_nested_try_stmt(&mut self) {
+        self.visiting_nested_try_stmt = false;
+    }
+
```

Comment:
Nit: These methods arent' realy recording anything. Rename to `enter_nested_try_block` and `exit_nested_try_block`